chain.go

```diff
@@ -50,7 +50,7 @@ func ChainIDfromDID(did w3c.DID) (ChainID, error) {
 func RegisterChainID(blockchain Blockchain, network NetworkID, chainID int) error {
 
 	k := fmt.Sprintf("%s:%s", blockchain, network)
-	if _, ok := chainIDs[k]; ok {
+	if existingChainID, ok := chainIDs[k]; ok && existingChainID != ChainID(chainID) {
```

Comment:
Hard to read. Could you please move `existingChainID, ok := chainIDs[k]` to separate line?